src/bcd/bcd/IDenoiser.h

```diff
@@ -13,15 +13,29 @@
 #ifndef I_DENOISER_H
 #define I_DENOISER_H
 
-#include <functional>
-#include <memory>
 
 namespace bcd
 {
 
 template<class T>
 class DeepImage;
 
+class IProgressReporter
+{
+  public:
+    IProgressReporter();
+    virtual ~IProgressReporter();
+
+    virtual void progress(const float i_progress) const = 0;
+
+    virtual bool isAborted() const = 0;
+
+  private:
+    // Non-copyable.
+    IProgressReporter(const IProgressReporter&);
```

Comment:
Make public and mark as `= delete` now that we're using C++11?

Reply:
I thought = delete was not supported by the VC version needed for some Max plugins.
Can I use it?

Reply:
Sorry, you're right, it's not supported by VS 2012.